### ISSUE TITLE
`make deploy` errors out

### ISSUE THREAD

**Issue by stevepiercy** [collective/cookiecutter-plone-starter#42]
Following the training successfully from start until it fails at this point:

https://training.plone.org/plone-deployment/deploy.html#using-provided-makefile

```console
$ make deploy
==> Deploy Stack in dev 
error during connect: Get "http://docker.example.com/v1.24/info": command [ssh -l plone -p 2222 -- 127.0.0.1 docker system dial-stdio] has exited with exit status 255, please make sure the URL is valid, and Docker 18.09 or later is installed on the remote host: stderr=plone@127.0.0.1: Permission denied (publickey).

make: *** [deploy] Error 1
```

I have Docker version 20.10.20.

I have no clue.

**Comment by stevepiercy** [comment]
FWIW, when I `@echo` out the command by tweaking the `Makefile` target, I get:

`docker --context dev stack deploy -c stacks/plone.yml plone`

I looked in that file, but nothing jumped out at me.

**Comment by stevepiercy** [comment]
One more data point, I use various named keys in my local `~/.ssh`, none of which are named `id_rsa`. I had pasted the value of `vagrant.pub` into `group_vars/all/users.yml` as specified in the training at [Configure SSH key](https://training.plone.org/plone-deployment/provision.html#configure-ssh-key). What is the expected value that should be pasted in there and what should it be named in `~/.ssh`?

**Comment by davisagli** [comment]
@stevepiercy It sounds like you've configured the VM to authorize access by the holder of the private key that is paired with the public key in `vagrant.pub`. If this is not your default private key, then you probably need to add explicit ssh configuration to use the right key when connecting to the VM. Something like this in `~/.ssh/config`, assuming the private key is in `~/.ssh/vagrant`:

```
Host 127.0.0.1
    IdentityFile ~/.ssh/vagrant
```

**Comment by davisagli** [comment]
@stevepiercy or if you don't want to change your ssh configuration, it looks like you can specify a key when creating the Docker context: https://stackoverflow.com/questions/63868735/docker-context-how-to-use-specific-ssh-key

**Comment by stevepiercy** [comment]
Revisiting this. Changing `~/.ssh/config` to the following worked for `make provision`:

```conf
Host 127.0.0.1
    IdentityFile ~/.ssh/vagrant
```

I'm more familiar with that technique. I can also toggle it, if it interferes with other development. Also the SO comments warn it might not work with Docker Compose, but I can always switch to it if needed.

I was able to proceed through [`make run-playbook`](https://training.plone.org/plone-deployment/provision.html#run-playbook), but now I don't know what to do next. I think that is more of a Training issue now, something that was said in person but not in writing 😢 .

Closing this issue here, and opening one in training. Thank you for your help @davisagli!
